website/src/docs/react.md

```diff
@@ -31,48 +27,55 @@ Instead of adding a UI plugin to an Uppy instance with `.use()`, the Uppy instan
 All other props are passed as options to the plugin.
 
 ```js
-import React from 'react'
+import React, { useEffect } from 'react'
 import Uppy from '@uppy/core'
-import Tus from '@uppy/tus'
-import { DragDrop } from '@uppy/react'
-
-const uppy = new Uppy({
-  meta: { type: 'avatar' },
-  restrictions: { maxNumberOfFiles: 1 },
-  autoProceed: true,
-})
-
-uppy.use(Tus, { endpoint: '/upload' })
-
-uppy.on('complete', (result) => {
-  const url = result.successful[0].uploadURL
-  store.dispatch({
-    type: 'SET_USER_AVATAR_URL',
-    payload: { url },
-  })
-})
-
-const AvatarPicker = ({ currentAvatar }) => {
-  return (
-    <div>
-      <img src={currentAvatar} alt="Current Avatar" />
-      <DragDrop
-        uppy={uppy}
-        locale={{
-          strings: {
-            // Text to show on the droppable area.
-            // `%{browse}` is replaced with a link that opens the system file selection dialog.
-            dropHereOr: 'Drop here or %{browse}',
-            // Used as the label for the link that opens the system file selection dialog.
-            browse: 'browse',
-          },
-        }}
-      />
-    </div>
-  )
+import { Dashboard } from '@uppy/react'
+
+const uppy = new Uppy()
+
+function Component () {
+  useEffect(() => {
+    return () => {
+      uppy.close({ reason: 'unmount' })
+    }
+  }, [])
+
+  return <Dashboard uppy={uppy} />
 }
 ```
 
+### Dynamic options
+
+```js
+import React, { useState, useEffect } from 'react'
+import Uppy from '@uppy/core'
+import { Dashboard } from '@uppy/react'
+
+const uppy = new Uppy()
+
+function Component ({ restrictions }) {
+  useEffect(() => {
+    // Change @uppy/core options
+    uppy.setOptions({ restrictions })
+
+    // Or change some plugin dynamically
+    uppy.getPlugin('SomePlugin').setOptions({ /* options */ })
+
+    return () => {
+      uppy.close({ reason: 'unmount' })
+    }
+  }, [restrictions])
+
+  return <Dashboard uppy={uppy} />
+}
```

Comment:
The Uppy instance here is shared between all instances of `Component` currently on the page. If I do:

```javascript
<Component restrictions={{allowedFileTypes: ['image/jpg']}} />

<Component restrictions={{allowedFileTypes: ['application/pdf']}} />
```

Won't they interfere with each other?

Reply:
Good point, in that case you want to do this:

```js
function Component ({ restrictions }) {
  const [uppy] = useState(() => new Uppy())

  useEffect(() => {
    uppy.setOptions({ restrictions })

    return () => {
      uppy.close({ reason: 'unmount' })
    }
  }, [restrictions])

  return <Dashboard uppy={uppy} />
}
```

This should also be added to the docs indeed. Thanks for pointing it out.